### PR TITLE
Stubs for arbitrary schema change added

### PR DESCRIPTION
**Pull request by efirs**
## Describe your changes

* Added calls to schema change code in update and read paths
* Refactored metadata code to better support schema versioning
* Added schema fixing and schema delta building stubs in schema/migration.go

## How best to test these changes

No functional change. Existing tests should cover.

## Issue ticket number and link

[TIG-749](https://tigrisdata.atlassian.net/browse/TIG-749)

[TIG-749]: https://tigrisdata.atlassian.net/browse/TIG-749?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: schema/migration.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+type Versions []Version
+
+func (x Versions) Len() int           { return len(x) }
+func (x Versions) Less(i, j int) bool { return x[i].Version < x[j].Version }
+func (x Versions) Swap(i, j int)      { x[i], x[j] = x[j], x[i] }
+
+func (x Versions) Latest() Version { return x[len(x)-1] }
+
+// VersionDeltaField describes field schema change.
+type VersionDeltaField struct {
```
Comment (himank) [inline schema/migration.go:38]: What if a field is dropped?

File: test/v1/server/document_test.go [hunk 2]
```diff
@@ -612,13 +612,13 @@ func TestInsert_AutoGenerated(t *testing.T) {
...
 func TestInsert_SchemaUpdate(t *testing.T) {
-	dbName := fmt.Sprintf("db_test")
+	dbName := fmt.Sprintf("db_test_%s", t.Name())
 
 	deleteProject(t, dbName)
 	createProject(t, dbName)
 	defer deleteProject(t, dbName)
 
-	collectionName := fmt.Sprintf("test_collection")
+	collectionName := fmt.Sprintf("test_collection_%s", t.Name())
```
[the remark this answers is not in the document]
Comment (efirs) [reply test/v1/server/document_test.go:621]: This is just simplifies debugging, easier to understand which test collection belongs too.

File: schema/migration.go (hunk 1 above)
Comment (efirs) [reply schema/migration.go:38]: If field is deleted then To is equal to `UnknownType`.
If field is added then From is equal to `Unknowntype`.

Comment (efirs) [reply schema/migration.go:41]: added a comment.

Comment (himank) [reply schema/migration.go:41]: if a field is added would it be part of DeltaFields?

Comment (efirs) [reply schema/migration.go:41]: It won't be. But just in case we need it, there is a way to express it.

**Review by himank — APPROVED** [review]

**Pull request merged**